filters/replicateto/replicateto.go

```diff
@@ -0,0 +1,140 @@
+package replicateto
+
+import (
+	"fmt"
+	"github.com/zalando/skipper/filters"
+	"net/http"
+	"net/url"
+	"regexp"
+	log "github.com/Sirupsen/logrus"
+)
+
+const (
+	Name = "replicateTo"
+)
+
+type replicateToSpec struct{}
+
+type replicateType int
+
+const (
+	asBackend replicateType = iota + 1
+	pathModified
+)
+
+type replicateTo struct {
+	client      *http.Client
+	typ         replicateType
+	host        string
+	scheme      string
+	rx          *regexp.Regexp
+	replacement []byte
+}
+
+// Returns a new replication filter Spec, whose instances execute
+// the exact same Request against a shadow backend.
+// parameters: shadow backend url, optional - the expression to match and the replacement string.
+// Name: "replicateTo".
+// Example
+// Path("/api/v3") -> replicateTo("https://api.example.com") -> "http://example.org/"
+// This route wil send incoming  request to http://example.org/api/v3 but will also send
+// a copy of the query to https://api.example.com/api/v3.
+// Example
+// Path("/api/v3") -> replicateTo("https://api.example.com", ".*", "/v1/" ) -> "http://example.org/"
+// This route wil send incoming request to http://example.org/api/v3 but will also send
+// a copy of the request to https://api.example.com/v1/ . Note that scheme and path are changed
+
+func NewReplicateTo() *replicateToSpec {
+	return &replicateToSpec{}
+}
+
+//We do not touch response at all
+func (r *replicateTo) Response(filters.FilterContext) {}
+
+// Request is copied and then modified to adopt changes in new backend
+func (r *replicateTo) Request(fc filters.FilterContext) {
+	copyOfRequest := cloneRequest(r, fc.Request())
+	go func() {
+		rsp, err := r.client.Do(&copyOfRequest)
+		if err != nil {
+			log.Warn("error while replicating request", err)
+		}
+		defer rsp.Body.Close()
+	}()
+}
+
+// copies requests changes URL and Host in request.
+// If 2nd and 3rd params are given path is also modified by applying regexp
+func cloneRequest(rep *replicateTo, req *http.Request) http.Request {
+	tmp := *req
```

Comment:
I'd follow the strategy of https://github.com/golang/go/blob/master/src/net/http/httputil/reverseproxy.go#L150 to clone the request

There seems to be a lot of (de)referencing going on here